editor/components/block-list/layout.js

```diff
@@ -320,4 +325,4 @@ export default connect(
 			dispatch( { type: 'REMOVE_BLOCKS', uids } );
 		},
 	} )
-)( BlockListLayout );
+)( withDragging( BlockListLayout ) );
```

Comment:
I feel like there's some indirection here? Can't we just apply this HoC to the real draggable element instead (here https://github.com/WordPress/gutenberg/pull/4115/files#diff-bc66d5701a2fea1111bb8f1bf0150910R577)

In theory, this could allow us to handle the `setTimeout` if they're necessary inside the HoC.

Any particular reason for this?

Reply:
Or, conversely, improve the higher-order component so it doesn't require the component to implement all of the relevant behavior. Is `withDragging` really reusable as-is outside of this specific usage?

Reply:
I completely see why there is confusion here. In fact, the HOC can just be applied to the block component directly. If my memory serves me well, there is no concept of a list anywhere in the drag handling logic. The HOC only makes available the two methods to produce a cross-browser drag image, move it around as the user drags the element, and discard it on drag end. That's all (and that is the only/best way I could device a solution that works equally across IE and the other browsers). 

If I am not mistaken (I will investigate closely on Saturday) this was left applied here because in my final few attempts to refactor the code into a separate component I had initially experimented with a "Draggable" list. I eventually failed and ended up with the HOC i describe above. Hence why I left this applied here. It could just wrap the block component instead and the effect would be the same - we just want those properties to become available to the block.

Reply:
p.s. please the read the HOC's README and let me know if something is not clear. The HOC only provides the methods to accomplish the sequence `clone and set a drag image -> move around with the mouse -> discard on drag end`. Nothing pertains to blocks specifically (see the arguments that are passed to the `onDragStart` closure). There is little styling involved too and no drop logic implemented - the latter is left to be implemented by the dropzone provider. So this can potentially be used to accomplish the same effect `clone and set a drag image -> move around with the mouse -> discard on drag end` in image galleries - or at least it's a step closer to that direction.